### PR TITLE
[ACA-1920] automate tests for Create file from template

### PR DESCRIPTION
**Pull request by adinapitul**
Issue Number: ACA-1920, ACA-2844

## Changes

- added components for Select Template and Create from Template dialogs
- added automated tests for Create File from Template action
- small refactorings

### CODE REVIEW

File: e2e/components/breadcrumb/dropdown-breadcrumb.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  constructor(ancestor?: string) {
+    super(DropDownBreadcrumb.selectors.root, ancestor);
+  }
+
+  async waitForPathListDropdownToOpen() {
+    await browser.wait(EC.presenceOf(this.pathItemsContainer), BROWSER_WAIT_TIMEOUT, 'Timeout waiting for breadcrumb dropdown to open');
+  }
+
+  async waitForPathListDropdownToClose() {
```
Comment (pionnegru) [inline e2e/components/breadcrumb/dropdown-breadcrumb.ts]: return type

File: e2e/components/breadcrumb/dropdown-breadcrumb.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  trigger: ElementFinder = this.component.element(by.css(DropDownBreadcrumb.selectors.trigger));
+  pathItems: ElementArrayFinder = browser.$$(DropDownBreadcrumb.selectors.pathOption);
+  pathItemsContainer: ElementFinder = browser.element(by.css('.mat-select-panel'));
+  currentFolder: ElementFinder = this.component.element(by.css(DropDownBreadcrumb.selectors.currentFolder));
+
+  constructor(ancestor?: string) {
+    super(DropDownBreadcrumb.selectors.root, ancestor);
+  }
+
+  async waitForPathListDropdownToOpen() {
```
Comment (pionnegru) [inline e2e/components/breadcrumb/dropdown-breadcrumb.ts]: return type

File: e2e/components/breadcrumb/dropdown-breadcrumb.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    await this.trigger.click();
+    await this.waitForPathListDropdownToOpen();
+  }
+
+  async clickPathItem(name: string): Promise<void> {
+    const elem = browser.element(by.cssContainingText(DropDownBreadcrumb.selectors.pathOption, name));
+    await elem.click();
+  }
+
+  async getPathItems() {
```
Comment (pionnegru) [inline e2e/components/breadcrumb/dropdown-breadcrumb.ts]: return type

File: e2e/components/data-table/data-table.ts [hunk 4]
```diff
@@ -150,15 +149,26 @@ export class DataTable extends Component {
     return this.body.all(by.css(DataTable.selectors.row));
   }
 
-  async countRows() {
+  async getRowsCount() {
```
Comment (pionnegru) [inline e2e/components/data-table/data-table.ts]: return type

File: e2e/components/dialog/create-from-template-dialog.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    await this.titleInput.clear();
+    await this.titleInput.sendKeys(title);
+  }
+
+  async enterDescription(description: string): Promise<void> {
+    await this.descriptionTextArea.clear();
+    await this.descriptionTextArea.sendKeys(description);
+  }
+
+  async deleteNameWithBackspace() {
```
Comment (pionnegru) [inline e2e/components/dialog/create-from-template-dialog.ts]: return type

File: e2e/components/dialog/select-template-dialog.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ElementFinder, by, browser, ExpectedConditions as EC } from 'protractor';
+import { BROWSER_WAIT_TIMEOUT } from '../../configs';
+import { Component } from '../component';
+import { DropDownBreadcrumb } from '../breadcrumb/dropdown-breadcrumb';
+// import { Utils } from '../../utilities/utils';
```
Comment (pionnegru) [inline e2e/components/dialog/select-template-dialog.ts]: remove if not used?

File: e2e/utilities/repo-client/apis/nodes/nodes-api.ts [hunk 7]
```diff
@@ -214,6 +224,24 @@ export class NodesApi extends RepoApi {
...
+      properties: {
+        'cm:destination': originalNodeId
+      }
+    }
+
+    try {
+      await this.apiAuth();
+      return await this.nodesApi.createNode(destinationId, nodeBody);
+    } catch (error) {
+      this.handleError(`${this.constructor.name} ${this.createNode.name}`, error);
```
Comment (pionnegru) [inline e2e/utilities/repo-client/apis/nodes/nodes-api.ts]: createNode doesn't return `NodeEntry` ? method should return Promise<NodeEntry | null> maybe?

File: e2e/utilities/repo-client/apis/nodes/nodes-api.ts [hunk 8]
```diff
@@ -325,6 +353,22 @@ export class NodesApi extends RepoApi {
   }
 
   // node permissions
+  async setInheritPermissions(nodeId: string, inheritPermissions: boolean) {
```
Comment (pionnegru) [inline e2e/utilities/repo-client/apis/nodes/nodes-api.ts]: return type

File: e2e/utilities/repo-client/apis/nodes/nodes-api.ts (hunk 7 above)
Comment (adinapitul) [reply e2e/utilities/repo-client/apis/nodes/nodes-api.ts]: nope, it returns Promise<NodeEntry| any>

**Pull request merged**